src/Symfony/Component/Console/Input/InputArgument.php

```diff
@@ -111,6 +113,19 @@ public function getDefault(): string|bool|int|float|array|null
         return $this->default;
     }
 
+    public function setCompletionValues(array|\Closure $completionValues = null)
+    {
+        $this->completionValues = $completionValues;
+    }
+
+    /**
+     * Returns suggestions for input completion.
+     */
+    public function getCompletionValues(): array|\Closure|null
+    {
+        return $this->completionValues;
```

Comment:
What about my previous proposal: calling the closure inside this method and always return array?

Reply:
I have the opposite optinion: 
- we don't care about the duplication, DRY is a bad principle
- not sure what you mean, why would this be an issue?
- adding the getter with an argument makes perfect sense to me. For the previous point, we might want to make it nullable (or not, to be investigated).

`public function getCompletionValues(CompletionInput): array` would make sense to me. It could event allow writting custom implementations of InputArgument with a dedicated completion method. Or am I missing something?
BTW, no need to make this able to return null, the empty array is just fine, isn't it?